### ISSUE TITLE
Future-proof `extism:host/env` namespace

### ISSUE THREAD

**Issue by chrisdickinson** [extism/extism#802]
Related to https://github.com/extism/extism/issues/801

Whenever we load a plugin, if we see that the module imports missing or mismatching items from `extism:host/env`, we can:

1. Emit a warning logline advising the user to install a newer version of the host sdk.
2. Stub the import with a failing host function.

Dynamic pdks (like js-pdk) would still be "backwards" compatible so long as newer host functionality isn't used.

**Comment by bhelx** [comment]
I'd like to think through the consequences of delaying the error like that. Should the plugin even run if it doesn't have all it's facilities? I think i can see a case for what you are suggesting. It makes the experience a bit smoother for the host. I just want to make sure we're not going to cause them way more pain down the line by not loudly stopping them here.

Also, if it's not possible to do this stub, i'd be fine with different behaviors on different SDKs. maybe some can outright error, some can try to fix it and warn. maybe recommend modsurfer? perhaps the xtp sdks could come with some kind of modsurfer integration?

**Comment by chrisdickinson** [comment]
> I'd like to think through the consequences of delaying the error like that. Should the plugin even run if it doesn't have all it's facilities? I think i can see a case for what you are suggesting. It makes the experience a bit smoother for the host. I just want to make sure we're not going to cause them way more pain down the line by not loudly stopping them here.

Yeah, I'm mostly thinking about dynamic language PDKs that import the entire namespace just in case the dynamic language portion of the plugin wants to use it. (This covers the js- and python-pdks today, but in the future it might include more?)

It might be a behavior we want hosts to be able to configure – whether to be lax about future extism features ("error-on-future-feature") or strict ("block-missing-features".)
